src/hooks/useFetch.js

```diff
@@ -56,19 +60,27 @@ export default function useFetch({
   );
 
   const error = formatError(result);
+  const statusCodeFromError = result?.error?.response?.status;
   useEffect(
     () => {
-      if (result?.status === 'loading') return;
-      setDisplayedError(error);
+      if (result?.status === 'loading') {
+        setDisplayedLoading(true);
+      } else {
+        if (statusCode !== statusCodeFromError)
+          setStatusCode(statusCodeFromError);
+        if (displayedError !== error) setDisplayedError(error);
+        setDisplayedLoading(false);
+      }
     },
-    [error, result?.status],
+    [error, result?.status, statusCodeFromError],
   );
 
   return {
     ...result,
     statusCode,
     data: dataAccessor(result),
-    loading: result?.isLoading,
+    isLoading: displayedLoading,
+    loading: displayedLoading,
```

Comment:
These are the same.

Reply:
yes, but some downstream components use one, some use the other. I was gonna organize it later.